### PR TITLE
Added listeners for new fliplet-like events

### PR DESCRIPTION
**Pull request by hcarneiro**
Because we rely on classes to show the bookmarked items, we need to make use of the new `liked.fail` & `unliked.fail` events to make sure the UI is updated correctly.

Changes made because of: https://github.com/Fliplet/fliplet-studio/issues/3735

### CODE REVIEW

File: css/layout-css/news-feed-style.upload.css [hunk 1]
```diff
@@ -1026,7 +1038,7 @@
   overflow-x: hidden;
   padding: 0 15px;
   margin: 54px 0 109px;
-  margin-bottom: calc(109px + env(safe-area-inset-bottom));
+  padding-bottom: env(safe-area-inset-bottom);
```
Comment (tonytlwu) [inline css/layout-css/news-feed-style.upload.css]: Is this to fix the issue we saw earlier on?

File: js/layout-javascript/agenda-code.js [hunk 2]
```diff
@@ -1693,33 +1682,27 @@ DynamicList.prototype.setupBookmarkButton = function(id, identifier, title) {
...
-}
 
-DynamicList.prototype.prepareSetupBookmarkOverlay = function(id) {
-  var _this = this;
-  var bookmarkIndentifier = id + '-bookmark';
-  var title = $('.agenda-detail-overlay').find('.agenda-item-inner-content .agenda-item-title').text();
+  if (button && button.btn) {
+    if (button.btn.isLiked()) {
+      $('.agenda-detail-overlay').find('.agenda-item-bookmark-holder-' + button.id).addClass('bookmarked');
+      isLiked = button.btn.isLiked()
```
Comment (tonytlwu) [inline js/layout-javascript/agenda-code.js]:
```suggestion
      isLiked = button.btn.isLiked();
```

File: js/layout-javascript/agenda-code.js [hunk 3]
```diff
@@ -1693,33 +1682,27 @@ DynamicList.prototype.setupBookmarkButton = function(id, identifier, title) {
...
+    } else {
+      $('.agenda-detail-overlay').find('.agenda-item-bookmark-holder-' + button.id).addClass('not-bookmarked');
+      isLiked = button.btn.isLiked()
```
Comment (tonytlwu) [inline js/layout-javascript/agenda-code.js]:
```suggestion
      isLiked = button.btn.isLiked();
```

File: js/layout-javascript/agenda-code.js [hunk 4]
```diff
@@ -1693,33 +1682,27 @@ DynamicList.prototype.setupBookmarkButton = function(id, identifier, title) {
...
+    }
+  } else {
+    $('.agenda-detail-overlay').find('.agenda-item-bookmark-holder').addClass('not-bookmarked');
+    isLiked = false
```
Comment (tonytlwu) [inline js/layout-javascript/agenda-code.js]:
```suggestion
    isLiked = false;
```

File: js/layout-javascript/news-feed-code.js [hunk 5]
```diff
@@ -792,19 +792,47 @@ DynamicList.prototype.deleteEntry = function(entryID) {
...
+    return btn.id === id;
+  });
+  var likeButton = _.find(_this.likeButtons, function(btn) {
+    return btn.id === id;
+  });
+
+  if (bookmarkButton && bookmarkButton.btn) {
+    if (bookmarkButton.btn.isLiked()) {
+      $('.news-feed-detail-overlay').find('.news-feed-bookmark-holder-' + bookmarkButton.id).addClass('bookmarked');
+      isBookmarked = bookmarkButton.btn.isLiked()
```
Comment (tonytlwu) [inline js/layout-javascript/news-feed-code.js]:
```suggestion
      isBookmarked = bookmarkButton.btn.isLiked();
```

File: js/layout-javascript/news-feed-code.js [hunk 6]
```diff
@@ -792,19 +792,47 @@ DynamicList.prototype.deleteEntry = function(entryID) {
...
+    } else {
+      $('.news-feed-detail-overlay').find('.news-feed-bookmark-holder-' + bookmarkButton.id).addClass('not-bookmarked');
+      isBookmarked = bookmarkButton.btn.isLiked()
+    }
+  } else {
+    $('.news-feed-detail-overlay').find('.news-feed-bookmark-holder').addClass('not-bookmarked');
+    isBookmarked = false
```
Comment (tonytlwu) [inline js/layout-javascript/news-feed-code.js]:
```suggestion
    isBookmarked = false;
```

File: js/layout-javascript/news-feed-code.js [hunk 7]
```diff
@@ -792,19 +792,47 @@ DynamicList.prototype.deleteEntry = function(entryID) {
...
   }
-  if (_this.data.social && (_this.data.social.bookmark || _this.data.social.likes)) {
-    _this.likesObserversOverlay(id);
+
+  if (likeButton && likeButton.btn) {
+    count = likeButton.btn.getCount() > 0 ? likeButton.btn.getCount() : '';
+    if (likeButton.btn.isLiked()) {
+      $('.news-feed-detail-overlay').find('.news-feed-like-holder-' + likeButton.id).addClass('liked');
+      $('.news-feed-detail-overlay').find('.news-feed-like-holder-' + likeButton.id + ' .count').html(count)
+      isLiked = likeButton.btn.isLiked()
```
Comment (tonytlwu) [inline js/layout-javascript/news-feed-code.js]:
```suggestion
      isLiked = likeButton.btn.isLiked();
```

File: js/layout-javascript/news-feed-code.js [hunk 8]
```diff
@@ -792,19 +792,47 @@ DynamicList.prototype.deleteEntry = function(entryID) {
...
+    } else {
+      $('.news-feed-detail-overlay').find('.news-feed-like-holder-' + likeButton.id).addClass('not-liked');
+      $('.news-feed-detail-overlay').find('.news-feed-like-holder-' + likeButton.id + ' .count').html(count)
+      isLiked = likeButton.btn.isLiked()
```
Comment (tonytlwu) [inline js/layout-javascript/news-feed-code.js]:
```suggestion
      isLiked = likeButton.btn.isLiked();
```

File: js/layout-javascript/news-feed-code.js [hunk 9]
```diff
@@ -850,83 +903,45 @@ DynamicList.prototype.likesObservers = function(from, to) {
...
 
-      if (button) {
-        button.btn.unlike();
-      }
+  $('.news-feed-detail-overlay').find('.news-feed-like-wrapper').on('click', function() {
+    if (isLiked) {
+      $(this).parents('.news-feed-like-holder').removeClass('liked').addClass('not-liked');
+      likeButton.btn.unlike();
+      $(this).find('.count').html(count);
+      isLiked = !isLiked
```
Comment (tonytlwu) [inline js/layout-javascript/news-feed-code.js]:
```suggestion
      isLiked = !isLiked;
```

File: js/layout-javascript/news-feed-code.js [hunk 10]
```diff
@@ -850,83 +903,45 @@ DynamicList.prototype.likesObservers = function(from, to) {
...
-    });
-
-    _this.likeButtonOverlay.on('unliked', function(data){
-      var entryTitle = this.$btn.parents('.news-feed-item-inner-content').find('.news-feed-item-title').text();
-      var button = _.find(_this.likeButtons, function(btn) {
-        return btn.id === id;
-      });
+    $(this).parents('.news-feed-bookmark-holder').removeClass('not-bookmarked').addClass('bookmarked');
+    bookmarkButton.btn.like();
+    isBookmarked = !isBookmarked
```
Comment (tonytlwu) [inline js/layout-javascript/news-feed-code.js]:
```suggestion
    isBookmarked = !isBookmarked;
```

File: js/layout-javascript/news-feed-code.js [hunk 11]
```diff
@@ -850,83 +903,45 @@ DynamicList.prototype.likesObservers = function(from, to) {
...
-DynamicList.prototype.likesObserversOverlay = function(id) {
+DynamicList.prototype.likesObserversOverlay = function(id, bookmarkButton, isBookmarked, likeButton, isLiked) {
   var _this = this;
+  var count;
+
+  $('.news-feed-detail-overlay').find('.news-feed-bookmark-wrapper').on('click', function() {
+    if (isBookmarked) {
+      $(this).parents('.news-feed-bookmark-holder').removeClass('bookmarked').addClass('not-bookmarked');
+      bookmarkButton.btn.unlike();
+      isBookmarked = !isBookmarked
```
Comment (tonytlwu) [inline js/layout-javascript/news-feed-code.js]:
```suggestion
      isBookmarked = !isBookmarked;
```

File: js/layout-javascript/news-feed-code.js [hunk 12]
```diff
@@ -850,83 +903,45 @@ DynamicList.prototype.likesObservers = function(from, to) {
...
-        category: 'list_dynamic_' + _this.data.layout,
-        action: 'entry_unbookmark',
-        label: entryTitle
-      });
-    });
-  }
+    $(this).parents('.news-feed-like-holder').removeClass('not-liked').addClass('liked');
+    likeButton.btn.like();
+    $(this).find('.count').html(count);
+    isLiked = !isLiked
```
Comment (tonytlwu) [inline js/layout-javascript/news-feed-code.js]: Too many. Find them and fix them.

File: css/layout-css/news-feed-style.upload.css (hunk 1 above)
Comment (hcarneiro) [reply css/layout-css/news-feed-style.upload.css]: @tonytlwu Yes!

**Pull request merged**